src/lib.rs

```diff
@@ -264,6 +264,23 @@ pub struct PackageInfo {
     pub cargo_package_id: Option<guppy::PackageId>,
 }
 
+impl PackageInfo {
+    /// Returns a web version of the repository URL,
+    /// with .git stripped if necessary.
+    pub fn web_url(&self) -> Option<String> {
```

Comment:
you could rewrite this with 1 less layer of `Option` encapsulation (rust has pretty nice sugar for this): https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=2629cfc9654e926cb95f17053b09c927